addons/api/addon/handlers/indexed-db-handler.js

```diff
@@ -0,0 +1,119 @@
+import { modelIndexes } from 'api/services/indexed-db';
+import { inject as service } from '@ember/service';
+import { getOwner, setOwner } from '@ember/application';
+import { pluralize } from 'ember-inflector';
+
+/**
+ * Handler to sit in front of the API request layer
+ * so we can handle any caching with indexedDb first
+ */
+export default class IndexedDbHandler {
+  @service indexedDb;
+
+  constructor(context) {
+    setOwner(this, getOwner(context));
+  }
+
+  async request(context, next) {
+    switch (context.request.op) {
+      case 'query': {
+        const { store, data } = context.request;
+        const { type, query, options: { pushToStore = true } = {} } = data;
+        const supportedModels = Object.keys(modelIndexes);
```

Comment:
Are there models that we don't support currently?

Reply:
I'm just starting with a whitelist, we technically can support all of them but this way we can start with only the models we want to use